src/common/com/intellij/plugins/haxe/ide/index/HaxeSymbolIndex.java

```diff
@@ -0,0 +1,183 @@
+import com.intellij.plugins.haxe.util.HaxeResolveUtil;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiManager;
+import com.intellij.psi.search.GlobalSearchScope;
+import com.intellij.psi.search.PsiElementProcessor;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.util.ArrayUtil;
+import com.intellij.util.CommonProcessors;
+import com.intellij.util.indexing.*;
+import com.intellij.util.io.EnumeratorStringDescriptor;
+import com.intellij.util.io.KeyDescriptor;
+import gnu.trove.THashMap;
+import gnu.trove.THashSet;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+
+public class HaxeSymbolIndex extends ScalarIndexExtension<String> {
+  public static final ID<String, Void> HAXE_SYMBOL_INDEX = ID.create("HaxeSymbolIndex");
+  private static final int INDEX_VERSION = HaxeIndexUtil.BASE_INDEX_VERSION + 8;
+  private DataIndexer<String, Void, FileContent> myDataIndexer = new MyDataIndexer();
+
+  @NotNull
+  @Override
+  public ID<String, Void> getName() {
+    return HAXE_SYMBOL_INDEX;
+  }
+
+  @Override
+  public int getVersion() {
+    return INDEX_VERSION;
+  }
+
+  @NotNull
+  @Override
+  public DataIndexer<String, Void, FileContent> getIndexer() {
+    return myDataIndexer;
+  }
+
+  @NotNull
+  @Override
+  public KeyDescriptor<String> getKeyDescriptor() {
+    return new EnumeratorStringDescriptor();
+  }
+
+  @NotNull
+  @Override
+  public FileBasedIndex.InputFilter getInputFilter() {
+    return HaxeSdkInputFilter.INSTANCE;
+  }
+
+  @Override
+  public boolean dependsOnFileContent() {
+    return true;
+  }
+
+  public static String[] getAllSymbols(@NotNull final GlobalSearchScope scope) {
+    final CommonProcessors.CollectProcessor<String> processor = new CommonProcessors.CollectProcessor<String>();
+    FileBasedIndex.getInstance().processAllKeys(HAXE_SYMBOL_INDEX, processor, scope, null);
+    return ArrayUtil.toStringArray(processor.getResults());
+  }
+
+  public static List<HaxeComponentName> getItemsByName(@NotNull final String name,
+                                                       @NotNull final Project project,
+                                                       @NotNull final GlobalSearchScope searchScope) {
+    final Collection<VirtualFile> files =
+      FileBasedIndex.getInstance().getContainingFiles(HAXE_SYMBOL_INDEX, name, searchScope);
+    final Set<HaxeComponentName> result = new THashSet<HaxeComponentName>();
+    for (VirtualFile vFile : files) {
+      final PsiFile psiFile = PsiManager.getInstance(project).findFile(vFile);
+      if (psiFile == null || psiFile.getFileType() != HaxeFileType.HAXE_FILE_TYPE) {
+        continue;
+      }
+      processComponents(psiFile, new PsiElementProcessor<HaxeNamedComponent>() {
+        @Override
+        public boolean execute(@NotNull HaxeNamedComponent subComponent) {
+          if (name.equals(subComponent.getName())) {
```

Comment:
But it was not available just because I forgot to add class names to indexer :)
And also to test this functionality, after update you need go to File-> Invalidate Cached / Restart...

After commit:
![image](https://cloud.githubusercontent.com/assets/3038174/13453787/d47125e6-e063-11e5-8d90-7c7a6846361c.png)